### PR TITLE
Merge intervals

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**
#4 
Time complexity = O(n^2)
Space complexity = O(n)

### CODE REVIEW

**Comment by DzyubSpirit** [comment]
Is it possible to reduce the time complexity to O(n * log(n))?

**Comment by JuliaGerasymenko** [comment]
Yes. You're right? I should just change the sorting algorithm with time complexity O(N log N)

**Comment by DzyubSpirit** [comment]
Yeah. right. That's would be the best implementation, I think.

File: mergeIntervals.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  for (let i = 0; i < intervals.length; i++) {
+    for (let j = i + 1; j < intervals.length; j++) {
+      if (intervals[i][0] > intervals[j][0]) {
+        [intervals[i], intervals[j]] = [intervals[j], intervals[i]];
+      }
+    }
+  }
+
+  for (let i = 0; i < len - 1; i++) {
+    if (
```
Comment (DzyubSpirit) [inline mergeIntervals.js]: This solution works and that's fine. But if you're interested, I think we can make it simpler.
Try do it this way:
```javascript
let res = [intervals.shift()];
for (let interval of intervals) {
...
}
```
I think there would less if-else branches.

**Comment by JuliaGerasymenko** [comment]
Does it was yours the simplest solution? Could you show me yours idea, please🙂

**Comment by JuliaGerasymenko** [comment]
image
also, it looks strange, because implementing another sorting algorithm didn't change the situation.

**Comment by JuliaGerasymenko** [comment]
I forgot, I already did it in the previous commit

File: mergeIntervals.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    for (let j = i + 1; j < intervals.length; j++) {
+      if (intervals[i][0] > intervals[j][0])
+        [intervals[i], intervals[j]] = [intervals[j], intervals[i]];
+    }
+  }
+
+  let res = [intervals.shift()];
+  for (let interval of intervals) {
+    let lastIndex = res.length - 1;
+    interval !== res[lastIndex] && interval[0] <= res[lastIndex][1]
```
Comment (DzyubSpirit) [inline mergeIntervals.js]: When does `interval === res[lastIndex]`?

**Review by DzyubSpirit — APPROVED** [review]

**Pull request merged**